lib/systems/examples.nix

```diff
@@ -297,6 +297,19 @@ rec {
     platform = {};
   };
 
+  #
+  # UEFI
+  #
+
+  # FIXME: this system double data may be disputed.
```

Comment:
The double from this I think is just `x86_64-windows`, which is fine.

Reply:
I think @amjoseph-nixpkgs didn't agree with this, but I suggest to take the disagreement discussions to gnu-config anyway. **I** also think that windows is a sad marker for this and would prefer `uefi` to appear somewhere.

Reply:
I think `doubles` are lossy to the point of being rather an anti-pattern :), so I just like simple "take the arch, take the OS, be done with it" logic.

Reply:
The main thing they are for is Nix knowing what derivations can build where, but I don't think anyone has proposed *running Nix itself on UEFI!* :joy: so that is non-issue for now.

Reply:
(I personally did multiple times and am planning to do it at some point, but that's beyond this PR indeed.)

Reply:
@RaitoBezarius you did? Wouldn't we need to finish porting to Windows before that is possible? Is there even a file system?

Reply:
@Ericson2314 I personally did propose running Nix on UEFI and would put the work to tvix on Windows and there's a Simple Filesystem abstraction yes. There's many blockers but it's goal for me for stage0-uefi.

Reply:
does kexec pass the tpm eventlog to the new kernel?

Reply:
kexec passes whatever you ask it to pass.  you can even build an entire custom initrd for the second kernel, at runtime.

if you're using a tpm you should probably think carefully about your threat model and why you think you need it.  [TPMs do not protect against evil maid attacks](https://www.errno.fr/BypassingBitlocker.html).

Reply:
> > running Nix on UEFI
> 
> I'm still waiting for somebody to explain what advantages this has over using directbooted Linux `kexec()` as a bootloader (i.e. `petitboot`).
> 
> The pre-`kexec()` Linux kernel can already do all these things, and can run tvix -- today, with no further modifications.  I netboot my build cluster machines this way, it's great.  No PXE garbage, zeroconf instead of dhcp, etc etc. 

I think you're right that it's better to run Linux.
Nevertheless, this does not preclude making good use of UEFI systems.